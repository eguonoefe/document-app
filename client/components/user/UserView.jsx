import React from 'react';
import { Modal } from 'react-materialize';
import PropTypes from 'prop-types';
/**
 * User view component for viewing a User
 * Has update and delete modals for deleting and updating a User
 *
 * @param {Object} props { id, firstName, lastName, email, deleteUser }
 * @returns {String} HTML markup for the User view
 */
const UserView = ({ id, firstName, lastName, email, deleteUser }) => (
  <div className="card blue-grey darken-1">
    <div className="card-content white-text">
      <span className="card-title">{firstName} {lastName}</span>
      <div>{email}</div>
    </div>
    <div className="card-action">
      <div className="row">
        <div className="col l12 m12 s12">
          <Modal
            trigger={
              <a
                className="waves-effect btn button-design"
                data-target="passwordModal">
                Delete
                </a>}
            actions={
              <div>
                <button
                  className="btn btn-flat modal-action modal-close"
                  name={id}
                  onClick={deleteUser}>
                  Delete
                  </button>
                <button
                  className="btn btn-flat modal-action modal-close left">
                  Cancel
                  </button>
              </div>}>
            <div >
              <h5 className="center">
                Are you sure you want to delete the user</h5>
            </div>
          </Modal>
        </div>
      </div>
    </div>
  </div>
);

UserView.propTypes = {
  id: PropTypes.number.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  deleteUser: PropTypes.func.isRequired,
};

export default UserView;
