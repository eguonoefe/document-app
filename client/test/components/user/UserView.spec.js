/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import UserView from '../../../components/user/UserView';

configure({ adapter: new Adapter() });

jest.dontMock('../../../components/user/UserView');

describe('UserView Component', () => {
  test('should match the user view snapshot', () => {
    const deleteUser = jest.fn();
    const component = shallow(
      <UserView
        id={3}
        firstName={'Eguono'}
        lastName={'Efe'}
        email={'efe@gmail.com'}
        deleteUser={deleteUser} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
