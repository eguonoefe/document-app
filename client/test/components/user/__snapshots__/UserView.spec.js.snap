// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView Component should match the user view snapshot 1`] = `
<div
  className="card blue-grey darken-1"
>
  <div
    className="card-content white-text"
  >
    <span
      className="card-title"
    >
      Eguono
       
      Efe
    </span>
    <div>
      efe@gmail.com
    </div>
  </div>
  <div
    className="card-action"
  >
    <div
      className="row"
    >
      <div
        className="col l12 m12 s12"
      >
        <Modal
          actions={
            <div>
              <button
                className="btn btn-flat modal-action modal-close"
                name={3}
                onClick={[MockFunction]}
              >
                Delete
              </button>
              <button
                className="btn btn-flat modal-action modal-close left"
              >
                Cancel
              </button>
            </div>
          }
          bottomSheet={false}
          fixedFooter={false}
          id="modal-1"
          options={
            Object {
              "dismissible": true,
              "endingTop": "10%",
              "inDuration": 250,
              "onCloseEnd": null,
              "onCloseStart": null,
              "onOpenEnd": null,
              "onOpenStart": null,
              "opacity": 0.5,
              "outDuration": 250,
              "preventScrolling": true,
              "startingTop": "4%",
            }
          }
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal"
            >
              Delete
            </a>
          }
        >
          <div>
            <h5
              className="center"
            >
              Are you sure you want to delete the user
            </h5>
          </div>
        </Modal>
      </div>
    </div>
  </div>
</div>
`;
